### DISCUSSION TITLE
Help pls! I can't read collection.

### DISCUSSION THREAD

**Discussion by GalimzyanovDanil** [isar/isar#1122] [Help]
Please help me to understand.
The application has a collection with one SomeClass field. The application is installed, works and stores the data.
Then I added a field like List<SomeClass> to this collection. I install the new version on top of the old one and get an error when trying to read data from this collection.

<img width="863" alt="isar_common" src="https://user-images.githubusercontent.com/86564275/220571334-e5706692-b093-4730-95a3-2b39349ab582.png">

<img width="1278" alt="StackTrace" src="https://user-images.githubusercontent.com/86564275/220571370-c065595d-88b7-4111-85df-da0dc810fc9a.png">

Isar 3.0.1

**Comment by xanmankey** [comment]
I'm not an expert on Isar at all, but could I ask you to post your collection code as well? If I understand you right, you made a change to a collection and after regenerating your collection, your code for reading from the database isn't working? If that's the case, not sure I can help, but it'd be nice to have the context of the collection to get a little bit more insight into the problem

**Reply by GalimzyanovDanil** [reply]
From
```dart
class SomeCollection {
    final SomeClass a;
}

@Embedded()
class SomeClass {
    final int a;
}
```

To
```dart
class SomeCollection {
    final SomeClass a;
    final List<SomeClass> b;
}

@Embedded()
class SomeClass {
    final int a;
}
```

But if you add an List int instead of List SomeClass, there are no errors and everything works.

**Reply by xanmankey** [reply]
I've never actually used a list of embedded classes, so I wrote some quick code to experiment and see if I could reproduce the error. I didn't have any luck reproducing the error, but here's the code I tried:
collection
```dart
import 'package:isar/isar.dart';

part 'test.g.dart';

@collection
class School {
  Id id = Isar.autoIncrement;
  late List<Student> students;
}

@Embedded()
class Student {
  late int grade;
  late String name;
}
```
main.dart
```dart
import 'package:isar/isar.dart';

import 'collection/test.dart';

void main() async {
  await initialize();
  Isar db = await Isar.open([SchoolSchema]);
  print(await write(db));
  print(await read(db));
  return;
}

Future<void> initialize() async {
  await Isar.initializeIsarCore(download: true);
}

Future<bool> write(Isar db) async {
  try {
    await db.writeTxn(() async {
      await db.schools.put(School()
        ..students = [
          Student()
            ..name = "George"
            ..grade = 96,
          Student()
            ..name = "Fred"
            ..grade = 72
        ]);
    });
    return true;
  } catch (e) {
    print(e);
    return false;
  }
}

Future<List<School>> read(Isar db) async {
  try {
    return await db.schools.where().findAll();
  } catch (e) {
    print(e);
    return [];
  }
}
```
output
```dart
true
[Instance of 'School']
```
I'm not sure what the range error is all about... it seems to be related to either invalid indexing or data struct size constraints. Perhaps either of those exist in your read code? I assume once you made the change to your collection you also changed your read code accordingly (if it was necessary)? That's about all I can offer based on what I know right now, I hope some of that is helpful. Definitely stay tuned to some other issues and discussions like the one at https://github.com/isar/isar/issues/1093#issue-1579132656, because it seems like some people are having similar issues, although I'm not sure if they're isar-related or not.
